examples/suit_update/Makefile

```diff
@@ -0,0 +1,117 @@
+# name of your application
+APPLICATION = suit_update
+
+# If no BOARD is found in the environment, use this default:
+BOARD ?= samr21-xpro
+
+# This has to be the absolute path to the RIOT base directory:
+RIOTBASE ?= $(CURDIR)/../..
+
+BOARD_INSUFFICIENT_MEMORY := arduino-duemilanove arduino-mega2560 arduino-nano \
+                             arduino-uno chronos msb-430 msb-430h \
+                             nucleo-f031k6 nucleo-f042k6 nucleo-l031k6 \
+                             nucleo-f030r8 nucleo-f303k8 nucleo-l053r8 \
+                             stm32f0discovery telosb waspmote-pro wsn430-v1_3b \
+                             wsn430-v1_4 z1
+
+# lower pktbuf size a little
+CFLAGS += -DGNRC_PKTBUF_SIZE=2000
+
+#
+# Networking
+#
+# Include packages that pull up and auto-init the link layer.
+# NOTE: 6LoWPAN will be included if IEEE802.15.4 devices are present
+USEMODULE += gnrc_netdev_default
+USEMODULE += auto_init_gnrc_netif
+# Specify the mandatory networking modules for IPv6 and UDP
+USEMODULE += gnrc_ipv6_router_default
+USEMODULE += gnrc_udp
+USEMODULE += gnrc_sock_udp
+# Additional networking modules that can be dropped if not needed
+USEMODULE += gnrc_icmpv6_echo
+
+# include this for printing IP addresses
+USEMODULE += shell_commands
+
+# Comment this out to enable code in RIOT that does safety checking
+# which is not needed in a production environment but helps in the
+# development process:
+#DEVELHELP = 1
+
+# Change this to 0 show compiler invocation lines by default:
+QUIET ?= 1
+
+#
+# SUIT update specific stuff
+#
+
+USEMODULE += nanocoap_sock sock_util
+USEMODULE += suit suit_coap
+
+# SUIT draft v4 support:
+USEMODULE += suit_v4
+
+# SUIT depends on riotboot support and some extra riotboot modules
+FEATURES_REQUIRED += riotboot
+USEMODULE += riotboot_slot riotboot_flashwrite riotboot_flashwrite_verify_sha256
+
+# Change this to 0 to not use ethos
+USE_ETHOS ?= 1
+
+ifeq (1,$(USE_ETHOS))
+  GNRC_NETIF_NUMOF := 2
+  USEMODULE += stdio_ethos
+  USEMODULE += gnrc_uhcpc
+
+  # ethos baudrate can be configured from make command
+  ETHOS_BAUDRATE ?= 115200
+  CFLAGS += -DETHOS_BAUDRATE=$(ETHOS_BAUDRATE)
+
+  # make sure ethos and uhcpd are built
+  TERMDEPS += host-tools
+
+  # For local testing, run
+  #
+  #     $ cd dist/tools/ethos; sudo ./setup_network.sh riot0 2001:db8::0/64
+  #
+  #... in another shell and keep it running.
+  export TAP ?= riot0
+  TERMPROG = $(RIOTTOOLS)/ethos/ethos
```

Comment:
There is an issue where if I kill the setup script before the terminal the tap interface doesn't get cleaned.

```
sudo dist/tools/ethos/setup_network.sh riot0 2001:db8::/64
net.ipv6.conf.riot0.forwarding = 1
net.ipv6.conf.riot0.accept_ra = 0
Joining IPv6 multicast group...
entering loop...
got packet from fe80::2 port 12345
got packet from fe80::2 port 12345
got packet from fe80::2 port 12345
got packet from fe80::2 port 12345
got packet from fe80::2 port 12345
^CCleaning up...
ioctl(TUNSETIFF): Device or resource busy

```

Reply:
uff, there's not much to do about this. the tap interface is still open.
We *could* hack something with lsof, but that would be ugly and non-portable...
Alternatively we could try removing the tap in a loop, like

```
while ip tuntab mode tap del $tap ; do
  echo "tap busy, ethos still running?"
  sleep 1
done
```

Do you think we need to solve this before PRing?

Reply:
No, lets not solve it afterwards